### PR TITLE
IA-1433: provide icon to reset search bar

### PR DESCRIPTION
**Pull request by codeKameleon**
[recording-2023-01-06-14-50-28.webm](https://user-images.githubusercontent.com/25134301/211025373-8b2072c0-f56a-40d4-a26f-e1e6d1461764.webm)

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -6657,7 +6657,8 @@
         "node_modules/fastest-levenshtein": {
             "version": "1.0.12",
             "resolved": "https://registry.npmjs.org/fastest-levenshtein/-/fastest-levenshtein-1.0.12.tgz",
-            "integrity": "sha512-On2N+BpYJ15xIC974QNVuYGMOlEVt4s0EOI3wwMqOmK1fdDY+FN/zltPV8vosq4ad4c/gJ1KHScUn/6AWIgiow=="
+            "integrity": "sha512-On2N+BpYJ15xIC974QNVuYGMOlEVt4s0EOI3wwMqOmK1fdDY+FN/zltPV8vosq4ad4c/gJ1KHScUn/6AWIgiow==",
```
Comment (beygorghor) [inline package-lock.json:6660]: i'm wondering why this one has changed ?

File: src/components/inputs/SearchInput/index.js [hunk 2]
```diff
@@ -23,6 +38,18 @@ const SearchInput = ({
...
+    };
+    const [hasClearIcon, setClearIcon] = useState(false);
+
+    useEffect(() => {
+        if (value !== '') {
+            setClearIcon(true);
+        } else {
+            setClearIcon(false);
+        }
+    }, [value]);
```
Comment (beygorghor) [inline src/components/inputs/SearchInput/index.js:52]: no need to use `useEffect` here, just create a `const` with a `useMemo` and remove `setClearIcon`

File: src/components/inputs/SearchInput/styles.js [hunk 3]
```diff
@@ -16,11 +16,17 @@ const styles = theme => ({
...
+        display: 'flex',
+        alignItems: 'center',
     },
     inputInput: {
         paddingRight: theme.spacing(7),
         width: '100%',
     },
+    searchIconWrapper: {
+        width: '24px',
+        height: '24px',
```
Comment (beygorghor) [inline src/components/inputs/SearchInput/styles.js:28]: `24px` = `theme.spacing(3)` 😉

**Review by beygorghor — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (codeKameleon) [reply package-lock.json:6660]: 🤔 i don't know

**Review by beygorghor — APPROVED** [review]

**Pull request merged**